Framework/DataHandling/src/LoadSampleShape.cpp

```diff
@@ -0,0 +1,222 @@
+#include "MantidDataHandling/LoadSampleShape.h"
+
+#include "MantidAPI/AnalysisDataService.h"
+#include "MantidAPI/FileProperty.h"
+#include "MantidAPI/InstrumentValidator.h"
+#include "MantidAPI/WorkspaceFactory.h"
+#include "MantidAPI/MatrixWorkspace.h"
+#include "MantidAPI/MatrixWorkspace_fwd.h"
+#include "MantidAPI/Sample.h"
+
+#include "MantidGeometry/Instrument.h"
+#include "MantidKernel/Material.h"
+#include "MantidKernel/CompositeValidator.h"
+#include "MantidKernel/EnabledWhenProperty.h"
+#include "MantidKernel/Exception.h"
+
+#include <Poco/File.h>
+
+namespace Mantid {
+namespace DataHandling {
+
+// Register the algorithm into the algorithm factory
+DECLARE_ALGORITHM(LoadSampleShape)
```

Comment:
You inherit from IFileLoader, but you only Declare_Algorithm, this does not make sense.  Do you understand how the IFileLoader classes work?